### ISSUE TITLE
Balance not updating

### ISSUE THREAD

**Issue by vikas-viki** [eth-infinitism/account-abstraction#319]
Hey before this commit `Allow SimpleAccount to transfer native token with executeBatch` everything was working fine but now when we deposit some eth the deposits is updating but not the balance 

![image](https://github.com/eth-infinitism/account-abstraction/assets/107193100/ba08f194-ea16-41e5-97a9-2685bc285f4d)

and when we send the eth using `withdrawDepositTo` the balance of reciever is updating but not the deposits even if the balance is 1 eth it is saying balance amount is too large 

![image](https://github.com/eth-infinitism/account-abstraction/assets/107193100/9f1d544c-1f11-46eb-9e27-1d7fa3a13529)

What should be done ?

# Previous version

Previously after calling `withdrawDepositTo` the reciever deposits was updating, below image is senders contract after sending deposits

![image](https://github.com/eth-infinitism/account-abstraction/assets/107193100/361fe896-d6c4-46d7-968f-6a508a472129)

the below one is the reciever contract, where the balance is not updated but deposits successfully came to contract.

![image](https://github.com/eth-infinitism/account-abstraction/assets/107193100/63cf1aa6-5a1b-4ab0-89af-ef21cb7f7568)

Is this the intended way or I'm missing something ?

**Comment by drortirosh** [comment]
Sorry, it is a bit confusing to read screen-shots. do you have commit ids or script that shows the problem ?
(or even account addresses and transaction-hashes, if it is on a public testnet)

**Comment by drortirosh** [comment]
We would like to help you with this issue, but we can't without further information.

**Comment by vikas-viki** [comment]
So let me explain,
Before the commit `Allow SimpleAccount to transfer native token with executeBatch`, what was happening is the Ethereum balance wasn't updating, like as you can see images in above previous version section, here is the flow

# Previous

Initially when I create a smart wallet and send some funds to it from EOA, the `ethers` balance wasn't updating, but the `deposits` variable is updating, as you can see from function call `getDeposit`. Here is pic

![accabst](https://github.com/eth-infinitism/account-abstraction/assets/107193100/baba563a-c29a-43a9-805e-227d4195e61a)

# Current

Steps:
1. Initially I created a wallet and sent some ethers (A wallet).
2. Again I created another wallet (B wallet).
3. I sent the funds from A to B.
Here the thing is in B wallet the `ethers` balance is updating but not the `deposits` variable. Here is the pic

![Frame 2](https://github.com/eth-infinitism/account-abstraction/assets/107193100/0bc6267b-a5b5-47b8-974e-247fc1df072d)

Here the problem is, in the previous version, before the commit mentioned above the **ethers** balance is not updating but the **deposits** variable is updating and we are able to send deposits from one wallet to another.
But now the **deposits** variable is not updating but the **ethers** balance is updating and we are not able to withdraw the deposits even if the balance is there. 

I hope you get this, sorry for replying late.

**Comment by drortirosh** [comment]
I don't think we've changed anything with the account regarding to deposit usage.
But I think you have some confusion about the deposit and its usage.

Some general notes about accounts:
- we're talking about account paying for itself, which means no paymaster is involved.
- An account can only pay from its "deposit" in the EntryPoint. the "getDeposit" method simply does `entryPoint.getDeposit(this)`
- If the deposit is not high enough, then when the validateUserOp is called for this UserOp,the account is given the amount it needs to deposit for itself. it should send (at minimum) this amount back to the entrypoint, otherwise the UserOp would fail.
- the above required deposit is the "maximum cost", and after the execution of the UserOperation, the excess is returned into the deposit.
- The account may withdraw its deposit (using `entryPoint.withdrawTo()` whenever it likes.

**Bottom line:**
- in general case, you shouldn't do anything with the deposit. Your account manages just enough deposit before each TX.
- for any eth transfer, please use your balance.

**Optimization:**
if you make a lot of transactions, and want to reduce (slightly) your gas costs, then manually send some eth from your account to the entrypoint. This way, you're not required to refresh the deposit on each UserOp.

**Comment by vikas-viki** [comment]
So the deposit meanse only the funds the wallet owner has deposited not the amount of ethers which it got from other person transferring to it then.

**Comment by drortirosh** [comment]
exactly. anything that was transfered to your account stays in your account, as your balance. you only move into a deposit what you want to be used to pay for gas.

**Comment by vikas-viki** [comment]
Yup
